controllers/validator.go

```diff
@@ -12,9 +12,34 @@ import (
 	"github.com/go-logr/logr"
 	oadpv1alpha1 "github.com/openshift/oadp-operator/api/v1alpha1"
 	"github.com/openshift/oadp-operator/pkg/credentials"
+
+	appsV1 "k8s.io/api/apps/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
 func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error) {
+	// When updating from OADP version 1.1.x to 1.2.x, delete
+	// daemonset.apps/restic (it was renamed to daemonset.apps/node-agent)
+	// resticrepositories.velero.io CRD (it was renamed to backuprepositories.velero.io)
+	if err := r.Delete(r.Context, &appsV1.DaemonSet{ObjectMeta: metaV1.ObjectMeta{
```

Comment:
I believe the Deletion of restic DS can be part of the `ReconcileNodeAgentDaemonset` function ?

Reply:
This one https://github.com/openshift/oadp-operator/blob/master/controllers/nodeagent.go#L72 ?

I believe is better for organization, yes :smile:

Reply:
lets do that then !